engine/runtime-integration-tests/src/test/java/org/enso/compiler/test/ExecCompilerTest.java

```diff
@@ -17,25 +17,27 @@
 import org.enso.common.MethodNames.Module;
 import org.enso.common.RuntimeOptions;
 import org.enso.compiler.core.ir.expression.errors.Conversion.DeclaredAsPrivate$;
+import org.enso.test.utils.ContextRule;
 import org.graalvm.polyglot.Context;
 import org.graalvm.polyglot.PolyglotException;
 import org.graalvm.polyglot.Source;
 import org.graalvm.polyglot.io.IOAccess;
 import org.hamcrest.core.AllOf;
 import org.junit.After;
 import org.junit.AfterClass;
-import org.junit.BeforeClass;
+import org.junit.ClassRule;
 import org.junit.Ignore;
 import org.junit.Test;
 
 public class ExecCompilerTest {
-  private static Context ctx;
-  private static ByteArrayOutputStream out;
+  @ClassRule
+  public static final ContextRule ctxRule =
+      ContextRule.createCustom(ExecCompilerTest::initEnsoContext);
```

Comment:
Rather then `createCustom` & other factory methods, get ready for [future evolution](http://wiki.apidesign.org/wiki/Evolution) and introduce builder. E.g
```suggestion
      ContextRule.newBuilder(ExecCompilerTest::initEnsoContext).build();
```
it will be easier to add new configuration options into a builder in the future.

Reply:
On the [latest revision](https://github.com/enso-org/enso/pull/12667/commits/281602e6a5700f4b0fe3c35b1c7aa733ea284e5e), ContextRule.Builder is already in https://github.com/enso-org/enso/blob/281602e6a5700f4b0fe3c35b1c7aa733ea284e5e/lib/java/test-utils/src/main/java/org/enso/test/utils/ContextRule.java#L205